src/utils/thread_helper.cpp

```diff
@@ -557,6 +552,39 @@ void ThreadHelper::AttachAllNodesTo(const std::vector<uint8_t> &aDatasetTlvs, Re
 
     datasetTlvs.mLength = reinterpret_cast<uint8_t *>(tlv->GetNext()) - datasetTlvs.mTlvs;
 
+    assert(datasetTlvs.mLength > 0);
+
+    if (role == OT_DEVICE_ROLE_DISABLED || role == OT_DEVICE_ROLE_DETACHED)
+    {
+        otOperationalDataset existingDataset;
+
+        error = otDatasetGetActive(mInstance, &existingDataset);
+        VerifyOrExit(error == OT_ERROR_NONE || error == OT_ERROR_NOT_FOUND);
+
+        if (error == OT_ERROR_NONE)
```

Comment:
better to use a flag to indicate migrate or not, so that we can avoid duplicate for enabling Thread.

Reply:
Done.